crates/node/src/lib.rs

```diff
@@ -852,10 +855,22 @@ impl Node {
             return Ok(None);
         };
 
-        let Some(context) = self.ctx_manager.get_context(&context_id)? else {
-            error!("Context not installed, but the transaction was in the pool.");
-            return Ok(None);
+        let handle = self.store.handle();
+
+        let Some(ctx_meta) = handle.get(&calimero_store::key::ContextMeta::new(
+            transaction.context_id,
+        ))?
+        else {
+            eyre::bail!("Context '{}' not found", transaction.context_id);
         };
+        let context = calimero_primitives::context::Context {
+            id: transaction.context_id,
+            application_id: ctx_meta.application_id.to_string().into(),
+        };
+
+        if ctx_meta.last_transaction_hash != *transaction.prior_hash {
```

Comment:
ah.. because of `last_transaction_hash`, why not add `last_transaction_hash` to `Context`?